### PR TITLE
Backend feature/crear grupo 159956714

### PR DESCRIPTION
**Pull request by joaquinvillar**
Create the model for group apply scaffold

### CODE REVIEW

File: api/app/models/group.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Group < ApplicationRecord
+  validates :name, presence: true
```
Comment (elcvilla) [inline api/app/models/group.rb]: Name must be uniqe

Comment (joaquinvillar) [reply api/app/models/group.rb]: Solved

File: api/db/schema.rb [hunk 2]
```diff
@@ -10,6 +10,54 @@
...
+
+  create_table "groups", force: :cascade do |t|
+    t.string "name"
+    t.string "description"
+    t.string "image"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "wye_workers", force: :cascade do |t|
```
Comment (LeticiaErrandonea) [inline api/db/schema.rb]: This doesn't seem like being part of this PR.

File: api/spec/controllers/groups_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# expectations of assigns and templates rendered. These features have been
+# removed from Rails core in Rails 5, but can be added back in via the
+# `rails-controller-testing` gem.
+
+RSpec.describe GroupsController, type: :controller do
+
+  # This should return the minimal set of attributes required to create a valid
+  # Group. As you add validations to Group, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) do
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: Why not `let(:valid_attributes) { ... }`?

File: api/spec/controllers/groups_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    { "name" => "testName", "description" => "testDescription", "image" => "url" }
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: Why not use `FactoryBot`?

File: api/spec/controllers/groups_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  end
+
+  # This should return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # GroupsController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "returns a success response" do
+      # group = Group.create! valid_attributes
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: Wye is this commented out?

File: api/spec/controllers/groups_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  describe "POST #create" do
+    context "with valid params" do
+      it "creates a new Group" do
+        expect do
+          post :create, params: { group: valid_attributes }, session: valid_session
+        end.to change(Group, :count).by(1)
+      end
+
+      it "renders a JSON response with the new group" do
+
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: nit: unnecessary extra newline

File: api/spec/controllers/groups_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        post :create, params: { group: valid_attributes }, session: valid_session
+        expect(response).to have_http_status(:created)
+        expect(response.content_type).to eq('application/json')
+        expect(response.location).to eq(group_url(Group.last))
+      end
+    end
+
+    context "with invalid params" do
+      it "renders a JSON response with errors for the new group" do
+
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: nit: unnecessary extra newline

File: api/spec/controllers/groups_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  describe "DELETE #destroy" do
+    it "destroys the requested group" do
+      group = Group.create! valid_attributes
+      expect do
+        delete :destroy, params: { id: group.to_param }, session: valid_session
+      end.to change(Group, :count).by(-1)
+    end
+  end
+
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: nit: unnecessary extra newline

File: api/spec/models/group_spec.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  subject do
+    described_class.new("name" => "testName", "description" => "testDescription", "image" => "url")
+  end
+  describe "Validations" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: nit: add newline between describe blocks

File: api/db/schema.rb (hunk 2 above)
Comment (joaquinvillar) [reply api/db/schema.rb]: Deleted.

File: api/spec/controllers/groups_controller_spec.rb (hunk 3 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Changed!

File: api/spec/controllers/groups_controller_spec.rb (hunk 4 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Added! nice helper.

File: api/spec/controllers/groups_controller_spec.rb (hunk 5 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Deleted, it was not necessary.

File: api/spec/controllers/groups_controller_spec.rb (hunk 6 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Deleted.

File: api/spec/controllers/groups_controller_spec.rb (hunk 7 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Deleted.

File: api/spec/controllers/groups_controller_spec.rb (hunk 8 above)
Comment (joaquinvillar) [reply api/spec/controllers/groups_controller_spec.rb]: Deleted.

File: api/spec/models/group_spec.rb (hunk 9 above)
Comment (joaquinvillar) [reply api/spec/models/group_spec.rb]: Added.

File: api/spec/controllers/groups_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+RSpec.describe GroupsController, type: :controller do
+
+  let(:valid_attributes) do
```
Comment (LeticiaErrandonea) [inline api/spec/controllers/groups_controller_spec.rb]: These `let`s seem to be doing more than should. Maybe you're looking for [this method](https://www.rubydoc.info/gems/factory_bot/FactoryBot%2FSyntax%2FMethods:attributes_for)? You can see how to use it in FactoryBot's [README](https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories).

File: api/spec/factories/groups.rb [hunk 11]
```diff
@@ -0,0 +1,21 @@
+FactoryBot.define do
+  factory :group do
+    trait :valid do
+      name { "Groupname" }
```
Comment (LeticiaErrandonea) [inline api/spec/factories/groups.rb]: nit: you can loose these braces

File: api/spec/models/group_spec.rb [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end
+
+  describe "Empty params" do
+    it "is not valid with an empty name" do
+      @group.name = nil
+      expect(@group).to_not be_valid
+    end
+
+    it "is not valid with an empty uid" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: This isn't testing what is said. Also, what is actually being testing should be failing (description being `nil` doesn't make a Group invalid).

File: api/spec/support/factory_bot.rb [hunk 13]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
```
Comment (LeticiaErrandonea) [inline api/spec/support/factory_bot.rb]: nit: newline missing here

File: api/app/models/group.rb [hunk 14]
```diff
@@ -1,3 +1,3 @@
 class Group < ApplicationRecord
-  validates :name, presence: true, uniqueness: true
+  validates :name, presence: { message: 'Group must have a name.' }, uniqueness: { message: 'Group name already taken' }
```
Comment (LeticiaErrandonea) [inline api/app/models/group.rb]: Active Record provides these kind of [error messages](https://guides.rubyonrails.org/active_record_validations.html#valid-questionmark-and-invalid-questionmark). Is there any reason these custom messages are needed?

File: api/spec/factories/groups.rb [hunk 15]
```diff
@@ -1,21 +1,15 @@
 FactoryBot.define do
   factory :group do
     trait :valid do
-      name "Groupname"
-      description "group description"
-      image "url"
+      name FFaker::Name.name
```
Comment (LeticiaErrandonea) [inline api/spec/factories/groups.rb]: 👏

File: api/spec/models/group_spec.rb [hunk 16]
```diff
@@ -1,26 +1,30 @@
 require 'rails_helper'
 
 RSpec.describe Group, type: :model do
+  describe "Name is being validated" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: I think my previous review message might have been confusing. I meant having this `describe` block nested in the `describe "Validations"` one

File: api/spec/models/group_spec.rb [hunk 17]
```diff
@@ -1,26 +1,30 @@
...
+    it "validate the presence" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: nit: `validates the presence`

File: api/spec/models/group_spec.rb [hunk 18]
```diff
@@ -1,26 +1,30 @@
...
+      should validate_presence_of(:name)
+        .with_message('Group must have a name.')
+    end
 
-  before(:all) do
-    @group = build(:group, :valid)
+    it "validate the uniqueness" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: nit: `validate the uniqueness`

File: api/spec/models/group_spec.rb [hunk 19]
```diff
@@ -1,26 +1,30 @@
 require 'rails_helper'
 
 RSpec.describe Group, type: :model do
+  describe "Name is being validated" do
+    it "validate the presence" do
+      should validate_presence_of(:name)
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: `expect` is preferred over `should`. This line should look like this `is_expected.to validate_presence_of(:name)`

File: api/spec/models/group_spec.rb [hunk 20]
```diff
@@ -1,26 +1,30 @@
...
+        .with_message('Group must have a name.')
+    end
 
-  before(:all) do
-    @group = build(:group, :valid)
+    it "validate the uniqueness" do
+      should validate_uniqueness_of(:name)
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: Same as above regarding `should` and `expect`

File: api/app/models/group.rb [hunk 21]
```diff
@@ -0,0 +1,3 @@
+class Group < ApplicationRecord
+  validates :name, presence: { message: 'Group must have a name.' }, uniqueness: { message: 'Group name already taken' }
```
Comment (LeticiaErrandonea) [inline api/app/models/group.rb]: Active Record provides these kind of [error messages](https://guides.rubyonrails.org/active_record_validations.html#valid-questionmark-and-invalid-questionmark). Is there any reason these custom messages are needed?

File: api/spec/factories/groups.rb [hunk 22]
```diff
@@ -0,0 +1,15 @@
+FactoryBot.define do
+  factory :group do
+    trait :valid do
+      name FFaker::Name.name
```
Comment (LeticiaErrandonea) [inline api/spec/factories/groups.rb]: 👏

File: api/spec/models/group_spec.rb [hunk 23]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  describe "Name is being validated" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: I think my previous review message might have been confusing. I meant having this `describe` block nested in the `describe "Validations"` one

File: api/spec/models/group_spec.rb [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+    it "validate the presence" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: nit: `validates the presence`

File: api/spec/models/group_spec.rb [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+      should validate_presence_of(:name)
+        .with_message('Group must have a name.')
+    end
+
+    it "validate the uniqueness" do
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: nit: `validates the uniqueness`

File: api/spec/models/group_spec.rb [hunk 26]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  describe "Name is being validated" do
+    it "validate the presence" do
+      should validate_presence_of(:name)
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: `expect` is preferred over `should`. This line should look more like this `is_expected.to validate_presence_of(:name)`

File: api/spec/models/group_spec.rb [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+        .with_message('Group must have a name.')
+    end
+
+    it "validate the uniqueness" do
+      should validate_uniqueness_of(:name)
```
Comment (LeticiaErrandonea) [inline api/spec/models/group_spec.rb]: Same as above regarding `should` and `expect`

File: api/spec/factories/groups.rb (hunk 22 above)
Comment (joaquinvillar) [reply api/spec/factories/groups.rb]: :+1:

File: api/spec/support/factory_bot.rb (hunk 13 above)
Comment (joaquinvillar) [reply api/spec/support/factory_bot.rb]: I dont know why i add the new line but dont appear here

File: api/.vscode/settings.json [hunk 28]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (LeticiaErrandonea) [inline api/.vscode/settings.json]: `.vscode` shouldn't be a part of the repository
